aderyn_core/src/detect/experimental/parent_chain.rs

```diff
@@ -30,11 +30,11 @@ impl IssueDetector for ParentChainDemonstrator {
             capture!(self, context, assignment);
 
             if let Some(parent_chain) = assignment.parent_chain(context) {
-                if let ASTNode::Block(block) = parent_chain[0] {
-                    capture!(self, context, block);
+                if let ASTNode::Block(_) = parent_chain[0] {
+                    capture!(self, context, parent_chain[0]);
                 }
-                if let ASTNode::ForStatement(for_statement) = parent_chain[1] {
-                    capture!(self, context, for_statement);
+                if let ASTNode::ForStatement(_) = parent_chain[1] {
+                    capture!(self, context, parent_chain[1]);
```

Comment:
Here we do not have to bother deriving `for_statement`. We just capture it in the `ASTNode` form : )